test/test_foreach.py

```diff
@@ -664,80 +655,64 @@ def test_bool_scalarlist(self, device, dtype):
+                        expected = [torch_bin_op(t, s) for t, s in zip(tensors, scalars)]
 
-                        with self.assertRaisesRegex(RuntimeError, "Subtraction, the `-` operator, with a bool tensor"):
-                            foreach_bin_op(tensors, scalars)
-                    else:
-                        if self.device_type == "cpu":
-                            expected = [torch_bin_op(t, s) for t, s in zip(tensors, scalars)]
-                            res = foreach_bin_op(tensors, scalars)
+                    with self.assertRaisesRegex(RuntimeError, "Subtraction, the `-` operator"):
+                        foreach_bin_op(tensors, scalars)
 
-                            self.assertEqual(res, expected)
+                    # There are a two types of different errors that will be thrown.
+                    # - Sub with bool is not allowed. 
+                    # - Result type can't be cast to the desired output type
+                    with self.assertRaises(RuntimeError):
+                        foreach_bin_op_(tensors, scalars)
+                    continue
 
-                            if dtype in torch.testing.integral_types() and foreach_bin_op_ == torch._foreach_div_:
-                                with self.assertRaisesRegex(RuntimeError, "result type Float can't be cast to the desired "):
-                                    foreach_bin_op_(tensors, scalars)
-                            else:
-                                foreach_bin_op_(tensors, scalars)
-                                self.assertEqual(tensors, expected)
-                        else:
-                            if foreach_bin_op == torch._foreach_sub:
-                                with self.assertRaisesRegex(RuntimeError, "Subtraction, the `-` operator, with a bo"):
-                                    expected = [torch_bin_op(t, s) for t, s in zip(tensors, scalars)]
-
-                                res = foreach_bin_op(tensors, scalars)
-                                foreach_bin_op_(tensors, scalars)
-                                self.assertEqual(res, tensors)
-                            else:
-                                expected = [torch_bin_op(t, s) for t, s in zip(tensors, scalars)]
-                                res = foreach_bin_op(tensors, scalars)
-
-                                if dtype in torch.testing.integral_types():
-                                    self.assertEqual(res, [e.to(dtype) for e in expected])
-                                else:
-                                    self.assertEqual(res, expected)
-
-                                foreach_bin_op_(tensors, scalars)
-                                self.assertEqual(res, tensors)
+                expected = [torch_bin_op(t, s) for t, s in zip(tensors, scalars)]
+                res = foreach_bin_op(tensors, scalars)
+                self.assertEqual(expected, res)
```

Comment:
This only tests correctness for the out-of-place variant, right? Should it also test correctness for the in-place variant?

Reply:
Done